### PR TITLE
User Profile Management

### PR DESCRIPTION
**Pull request by RobertCram**
PT Story: https://www.pivotaltracker.com/story/show/147882825

**Changes proposed in this pull request:**

1. Add first and last name fields to the Devise new registration and edit registration views.
2. Make first and last name required fields on user creation (before they were only required when editing a user)

**Note:**

This story does NOT include a UI link to the edit registration view. For that, see story  https://www.pivotaltracker.com/story/show/148847945

Ready for review:
@patmbolger @weedySeaDragon @thesuss

### CODE REVIEW

File: app/views/devise/registrations/edit.html.haml [hunk 1]
```diff
@@ -6,9 +6,17 @@
   = form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f|
     = devise_error_messages!
```
Comment (patmbolger) [inline app/views/devise/registrations/edit.html.haml]: We need to customize error treatment to be consistent with our other views.  (see similar changes in PR: https://github.com/AgileVentures/shf-project/pull/297/files)

In particular, this line should be replaced by:
```
if resource.errors.any?
  .wpcf7-response-output
    - resource.errors.full_messages.each do |msg|
      = msg
```

File: app/views/devise/registrations/new.html.haml [hunk 2]
```diff
@@ -6,9 +6,17 @@
   = form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f|
     = devise_error_messages!
```
Comment (patmbolger) [inline app/views/devise/registrations/new.html.haml]: Same comment as above.

**Review by patmbolger — CHANGES_REQUESTED** [review]
Please see comments.

File: app/views/devise/registrations/edit.html.haml (hunk 1 above)
Comment (RobertCram) [reply app/views/devise/registrations/edit.html.haml]: I've made the requested change, with one minor difference: I had to add a break after the individual messages. The code above as is would put multiple error messages on one line.

Comment (patmbolger) [reply app/views/devise/registrations/edit.html.haml]: Yup, thats how all of our validation errors appear now.  I have a story open to replace the error-handling code with a helper method that returns styled HTML for error display - we can fix this issue there.

Comment (RobertCram) [reply app/views/devise/registrations/edit.html.haml]: That's not what I am seeing. The errors on the membership application page seem to render correctly without the added breaks.

Comment (patmbolger) [reply app/views/devise/registrations/edit.html.haml]: Yes, for MA `create` we use this logic:
```
- if @membership_application.errors.any?
  - @membership_application.errors.full_messages.each do |msg|
    .wpcf7-response-output
      = msg
```
whereas in other places we use this kind of logic:
```
.wpcf7-response-output
  - @member_app_waiting_reason.errors.full_messages.each do |msg|
    = msg
```
The styling for `wpcf7-response-output` class includes `text-align:center;`.  If each message is separately styled (first example above) this forces each message to occur on a separate line.  The second example shows all messages on a single (centered) line.

We're all over the map here and thus we need to standardize this.

Also, here is another issue I just noticed: https://www.pivotaltracker.com/story/show/148990401

**Review by patmbolger — APPROVED** [review]

**Review by weedySeaDragon — APPROVED** [review]

**Pull request merged**